src/modules/fancyzones/editor/FancyZonesEditor/Models/LayoutModel.cs

```diff
@@ -219,30 +221,68 @@ public static ObservableCollection<LayoutModel> LoadCustomModels()
                     var info = current.GetProperty(InfoJsonTag);
                     if (type.Equals(GridJsonTag))
                     {
+                        bool error = false;
+
                         int rows = info.GetProperty(RowsJsonTag).GetInt32();
                         int columns = info.GetProperty(ColumnsJsonTag).GetInt32();
 
                         List<int> rowsPercentage = new List<int>(rows);
                         JsonElement.ArrayEnumerator rowsPercentageEnumerator = info.GetProperty(RowsPercentageJsonTag).EnumerateArray();
+                        if (rows <= 0 || columns <= 0 || rowsPercentageEnumerator.Count() != rows)
+                        {
+                            error = true;
+                        }
+
                         while (rowsPercentageEnumerator.MoveNext())

```

Comment:
`error` check in while loops can prevent from unnecessary work.

Reply:
done